### DISCUSSION TITLE
How to make a layout scrollable

### DISCUSSION THREAD

**Discussion by wasi-master** [Textualize/rich#1448] [Q&A] [answered]
So I've been using Layouts for a while now and they are great. the only problem I have right now is that sometimes my data does not fit inside a single layout, and rich seems to trim the text to make room for other stuff and to fit the layout to the screen. now my question is that is there is a way to have it print the full text and make it so that the user can just scroll to read the full text? I am not using Live

**TL;DR** I want to make a Layout scrollable

**Comment by willmcgugan** [comment, marked as answer]
Rich doesn't support scrolling. You could possibly use Textual for this.

**Reply by wasi-master** [reply]
Thanks, I will.

**Reply by wasi-master** [reply]
Although is there any way to do it with the terminal implementation (e.g. printing the entire thing at once)? or would I have to use panels instead?
